test/e2e/secretgencontroller/private_registry_auth_test.go

```diff
@@ -0,0 +1,260 @@
+	sas := e2e.ServiceAccounts{env.Namespace}
+
+	// If this changes, the skip-tls-verify domain must be updated to match
+	registryNamespace := "registry"
+	registryName := "test-registry"
+
+	pkgiYaml := fmt.Sprintf(`---
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  name: pkg.test.carvel.dev
+  namespace: %[1]s
+spec:
+  # This is the name we want to reference in resources such as PackageInstall.
+  displayName: "Test PackageMetadata in repo"
+  shortDescription: "PackageMetadata used for testing"
+  longDescription: "A longer, more detailed description of what the package contains and what it is for"
+  providerName: Carvel
+  maintainers:
+  - name: carvel
+  categories:
+  - testing
+  supportDescription: "Description of support provided for the package"
+---
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  name: pkg.test.carvel.dev.1.0.0
+  namespace: %[1]s
+spec:
+  refName: pkg.test.carvel.dev
+  version: 1.0.0
+  template:
+    spec:
+      fetch:
+      - imgpkgBundle:
+          image: registry-svc.%[3]s.svc.cluster.local:443/secret-test/test-bundle
+      template:
+      - ytt: {}
+      - kbld:
+          paths:
+          - "-"
+          - ".imgpkg/images.yml"
+      deploy:
+      - kapp: {}
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  name: %[2]s
+  namespace: %[1]s
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/packageinstalls
+spec:
+  syncPeriod: 30s
+  serviceAccountName: kappctrl-e2e-ns-sa
+  packageRef:
+    refName: pkg.test.carvel.dev
+    versionSelection:
+      constraints: 1.0.0
+`, env.Namespace, name, registryNamespace) + sas.ForNamespaceYAML()
+
+	secretYaml := fmt.Sprintf(`
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: regcred
+type: kubernetes.io/dockerconfigjson
+stringData:
+  .dockerconfigjson: |
+    {
+      "auths": {
+        "registry-svc.%s.svc.cluster.local:443": {
+          "username": "testuser",
+          "password": "testpassword",
+          "auth": ""
+        }
+      }
+    }
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: SecretExport
+metadata:
+  name: regcred
+spec:
+  toNamespaces:
+  - %s
+`, registryNamespace, env.Namespace)
+
+	cleanUp := func() {
+		// Delete App with kubectl first since kapp
+		// deletes ServiceAccount before App
+		kubectl.RunWithOpts([]string{"delete", "apps/" + name}, e2e.RunOpts{AllowError: true})
+		kapp.Run([]string{"delete", "-a", registryName, "-n", registryNamespace})
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", "secret-export"})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("deploy registry with self signed certs", func() {
+		kapp.Run([]string{"deploy", "-f", "../assets/registry/registry2.yml", "-f", "../assets/registry/certs-for-skip-tls.yml", "-f", "../assets/registry/htpasswd-auth", "-f", "../assets/registry/registry-contents.yml", "-a", registryName})
```

Comment:
any reason why we just dont use entire directory for -f, or combine all registry config into a single file?

Reply:
I think there may be additional test assets under that folder which are shared with other tests.